src/test/groovy/com/intuit/graphql/orchestrator/integration/TopLevelStitchingSpec.groovy

```diff
@@ -126,4 +129,91 @@ class TopLevelStitchingSpec extends BaseIntegrationTestSpecification {
         data.upsertProfile?.prefLastName instanceof String && data.upsertProfile?.prefLastName == "Last Name"
         data.upsertProfile?.version instanceof Integer && data.upsertProfile?.version == Short.MAX_VALUE
     }
+
+    def "Empty provider cannot stitch"() {
+        given:
+        def emptyProvider = TestServiceProvider.newBuilder()
+                .namespace("Empty")
+                .serviceType(ServiceProvider.ServiceType.GRAPHQL)
+                .sdlFiles(ImmutableMap.of())
+                .build()
+
+        when:
+        createGraphQLOrchestrator(emptyProvider)
+
+        then:
+        thrown(StitchingException.class)
+    }
+
+    def "Provider with empty file cannot stitch successfully"() {
+        given:
+        def emptyProvider = TestServiceProvider.newBuilder()
+                .namespace("Empty")
+                .serviceType(ServiceProvider.ServiceType.GRAPHQL)
+                .sdlFiles(ImmutableMap.of("schema.graphqls", ""))
+                .build()
+
+        when:
+        createGraphQLOrchestrator(emptyProvider)
+
+        then:
+        thrown(StitchingException.class)
+    }
+
+    def "Provider with empty operation and extending operation can stitch"() {
+        given:
+
+        String schema = "type Query {} extend type Query {getID: ID}"
```

Comment:
not sure if needed, should we add test for empty operation.. e.g. `type Query {}`, i.e. no `extend type Query`

Reply:
added a test case for that